test/preview/components/audio/test_whisper_remote.py

```diff
@@ -1,197 +1,236 @@
+from unittest.mock import patch
 import os
-from unittest.mock import MagicMock, patch
+
 
 import pytest
+import openai
+from openai.util import convert_to_openai_object
+
 
 from haystack.preview.dataclasses import Document
-from haystack.preview.components.audio.whisper_remote import RemoteWhisperTranscriber, OPENAI_TIMEOUT
+from haystack.preview.components.audio.whisper_remote import RemoteWhisperTranscriber
+
+
+def mock_openai_response(response_format="json", **kwargs) -> openai.openai_object.OpenAIObject:
+    if response_format == "json":
+        dict_response = {"text": "test transcription"}
+    # Currently only "json" is supported.
+    else:
+        dict_response = {}
+
+    return convert_to_openai_object(dict_response)
 
 
 class TestRemoteWhisperTranscriber:
     @pytest.mark.unit
-    def test_init_unknown_model(self):
-        with pytest.raises(ValueError, match="not recognized"):
-            RemoteWhisperTranscriber(model_name="anything", api_key="something")
+    def test_init_no_key(self, monkeypatch):
+        monkeypatch.delenv("OPENAI_API_KEY", raising=False)
+        error_msg = "RemoteWhisperTranscriber expects an OpenAI API key."
+        with pytest.raises(ValueError, match=error_msg):
+            RemoteWhisperTranscriber(api_key=None)
 
```

Comment:
We should also test that both ways to set the key (through env var and through the module global var) work, and which one wins if they're both set.